src/games/strategy/engine/framework/systemcheck/LocalSystemChecker.java

```diff
@@ -0,0 +1,68 @@
+package games.strategy.engine.framework.systemcheck;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.Set;
+
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+
+/**
+ * This class runs a set of local system checks, like access network, and create a temp file.
+ * Each check is always run, and this class records the results of those checks.
+ */
+public final class LocalSystemChecker {
+
+  private final Set<SystemCheck> systemChecks;
+
+  public LocalSystemChecker() {
+    this(ImmutableSet.of(defaultNetworkCheck(), defaultFileSystemCheck()));
+  }
+
+  private static final SystemCheck defaultNetworkCheck() {
+    return new SystemCheck("Can connect to github.com (check network connection)", () -> {
+      try {
+        URL url = new URL("http://www.github.com");
+        url.openConnection();
+      } catch (Exception e) {
+        Throwables.propagate(e);
+      }
+    });
+  }
+
+  private static final SystemCheck defaultFileSystemCheck() {
+    return new SystemCheck("Can create temporary files (check disk usage, file permissions)", () -> {
+      try {
+        File.createTempFile("prefix", "suffix");
+      } catch (IOException e) {
+        Throwables.propagate(e);
+      }
+    });
+  }
+
+
+  protected LocalSystemChecker(Set<SystemCheck> checks) {
```

Comment:
What the reason for having a protected constructor that is only called in one place by this class? Shouldn't this set just be done by the public constructor?